### ISSUE TITLE
Unresolvable dependency error only points to top of the stack

### ISSUE THREAD

**Issue by jtojnar** [slince/di#9]
When an error occurs, for example due to #8, the library will only report the direct dependency that failed:

```
Fatal error: Uncaught Slince\Di\Exception\DependencyInjectionException: Unresolvable dependency resolving "contentLoader" in class "Controller" in /home/jtojnar/Projects/selfoss/vendor/slince/di/Resolver.php:193
Stack trace:
#0 /home/jtojnar/Projects/selfoss/vendor/slince/di/Resolver.php(102): Slince\Di\Resolver->resolveDependencies(Array, Array)
#1 /home/jtojnar/Projects/selfoss/vendor/slince/di/Resolver.php(50): Slince\Di\Resolver->createFromClass(Object(Slince\Di\Definition))
#2 /home/jtojnar/Projects/selfoss/vendor/slince/di/Container.php(261): Slince\Di\Resolver->resolve(Object(Slince\Di\Definition))
#3 /home/jtojnar/Projects/selfoss/vendor/slince/di/Container.php(221): Slince\Di\Container->resolveInstance('Controller')
#4 /home/jtojnar/Projects/selfoss/slince.php(36): Slince\Di\Container->get('Controller')
#5 {main}
  thrown in /home/jtojnar/Projects/selfoss/vendor/slince/di/Resolver.php on line 193
```

```php
<?php

require_once __DIR__ . '/vendor/autoload.php';

class Logger { }

class FileStorage {
    public function __construct(Logger $logger, string $directory) { }
}

class IconStore {
    public function __construct(FileStorage $storage) { }
}

class ContentLoader {
    public function __construct(IconStore $iconStore) { }
}

class Controller {
    public function __construct(ContentLoader $contentLoader) { }
}

$container = new Slince\Di\Container();

$container
    ->register('$iconStorageBackend', FileStorage::class)
    ->addArgument(__DIR__ . '/favicons')
;

$container
    ->register(IconStore::class)
    ->addArgument(new Slince\Di\Reference('$iconStorageBackend'))
    ->setShared(true)
;

var_dump($container->get(Controller::class));
```

It would be more useful for diagnostics if the full chain of the failed dependencies could be reported.

**Comment by slince** [comment]
fixed in 3.2.3

**Comment by jtojnar** [comment]
Thanks, this works great.
